apps/api/prisma/schema.prisma

```diff
@@ -78,14 +78,138 @@ model User {
-  lastLoginAt         DateTime? @map("last_login_at")
+  address     String    @unique @map("address") @db.VarChar(42)
+  email       String?   @unique
+  phoneNumber String?   @unique @map("phone_number")
+  banned      Boolean   @default(false)
+  comment     String?
+  lastLoginAt DateTime? @map("last_login_at")
+
+  // Relations
+  profile              UserProfile?
+  socialAccounts       UserSocialAccount[]
+  notificationSettings UserNotificationSettings[]
 
   // Computed
   createdAt DateTime @default(now()) @map("created_at")
   updatedAt DateTime @updatedAt @map("updated_at")
 
   @@map("user")
 }
+
+model UserProfile {
+  // Primary key
+  id String @id @default(cuid())
+
+  // Fields
+  // TreasureTag
+  tag              String?
+  discriminant     Int?      @db.SmallInt
+  tagClaimed       Boolean   @default(false) @map("tag_claimed")
+  tagModifiedAt    DateTime? @map("tag_modified_at")
+  tagLastCheckedAt DateTime? @map("tag_last_checked_at")
+
+  // Email security
+  emailSecurityPhrase          String?   @map("email_security_phrase")
+  emailSecurityPhraseUpdatedAt DateTime? @map("email_security_phrase_updated_at")
+
+  // Bio
+  featuredNftIds        String[] @map("featured_nft_ids")
+  featuredBadgeIds      String[] @map("featured_badge_ids")
+  highlyFeaturedBadgeId String?  @map("highly_featured_badge_id")
+  about                 String?
+  pfp                   String?
+  banner                String?
+
+  // UI
+  showMagicBalance Boolean @default(true) @map("show_magic_balance")
+  showEthBalance   Boolean @default(true) @map("show_eth_balance")
+  showGemsBalance  Boolean @default(true) @map("show_gems_balance")
+
+  // Migration
+  legacyAddress String? @unique @map("legacy_address") @db.VarChar(42)
+
+  // Other
+  testnetFaucetLastUsedAt DateTime? @map("testnet_faucet_last_used_at")
+
+  // Relations
+  userId String? @unique @map("user_id")
+  user   User?   @relation(fields: [userId], references: [id])
+
+  // Computed
+  createdAt DateTime @default(now()) @map("created_at")
+  updatedAt DateTime @updatedAt @map("updated_at")
+
+  @@unique([tag, discriminant])
+  @@map("user_profile")
+}
+
+enum SocialNetwork {
+  APPLE
+  DISCORD
+  FARCASTER
+  GOOGLE
+  STEAM
+  TWITCH
+  TWITTER
+}
+
+model UserSocialAccount {
+  // Primary key
+  id String @id @default(cuid())
+
+  // Fields
+  network       SocialNetwork
+  accountId     String        @map("account_id")
+  accountName   String        @map("account_name")
+  accountHandle String?       @map("account_handle")
+  accountData   Json?         @map("account_data")
+  isPublic      Boolean       @default(false) @map("is_public")
+
+  // Migration
+  legacyAddress String? @unique @map("legacy_address") @db.VarChar(42)
+
+  // Relations
+  userId String? @map("user_id")
+  user   User?   @relation(fields: [userId], references: [id])
+
+  // Computed
+  createdAt DateTime @default(now()) @map("created_at")
+  updatedAt DateTime @updatedAt @map("updated_at")
+
```

Comment:
Does there need to be a `@@unique([userId, network])` or something similar here? That is, a user can have at most one Farcaster associated? Or will we allow users to have more than one of each type associated? Let's say I have multiple Farcaster accounts. Do i assocated all of them with this account? Or at most one of them with this account?

Reply:
Thought about this a bit as well and landed on just not restricting it. I figure some users may have multiple accounts they want to associate (e.g., their personal and company profiles)